src/braket/pennylane_plugin/translation.py

```diff
@@ -14,40 +14,71 @@
 from functools import singledispatch
 from typing import FrozenSet, Type
 
+import numpy as np
 import pennylane as qml
 from braket.circuits import Gate, ResultType, gates, observables
 from braket.circuits.result_types import Expectation, Probability, Sample, Variance
 from pennylane.operation import Observable, ObservableReturnTypes, Operation
 
 _OPERATION_MAP = {
+    # pennylane operations
     "Hadamard": gates.H,
+    "Hadamard.inv": gates.H,
     "PauliX": gates.X,
+    "PauliX.inv": gates.X,
     "PauliY": gates.Y,
+    "PauliY.inv": gates.Y,
     "PauliZ": gates.Z,
-    "CNOT": gates.CNot,
-    "CY": gates.CY,
-    "CZ": gates.CZ,
+    "PauliZ.inv": gates.Z,
     "S": gates.S,
+    "S.inv": gates.Si,
     "T": gates.T,
-    "V": gates.V,
+    "T.inv": gates.Ti,
+    "SX": gates.V,
+    "SX.inv": gates.Vi,
+    "CNOT": gates.CNot,
+    "CNOT.inv": gates.CNot,
+    "CZ": gates.CZ,
+    "CZ.inv": gates.CZ,
+    "CY": gates.CY,
+    "CY.inv": gates.CY,
+    "SWAP": gates.Swap,
+    "SWAP.inv": gates.Swap,
+    "CSWAP": gates.CSwap,
+    "CSWAP.inv": gates.CSwap,
+    "Toffoli": gates.CCNot,
+    "Toffoli.inv": gates.CCNot,
+    "RX": gates.Rx,
+    "RX.inv": gates.Rx,
+    "RY": gates.Ry,
+    "RY.inv": gates.Ry,
+    "RZ": gates.Rz,
+    "RZ.inv": gates.Rz,
     "PhaseShift": gates.PhaseShift,
+    "PhaseShift.inv": gates.PhaseShift,
+    "QubitUnitary": gates.Unitary,
+    "QubitUnitary.inv": gates.Unitary,
+    # plugin operations
     "CPhaseShift": gates.CPhaseShift,
+    "CPhaseShift.inv": gates.CPhaseShift,
     "CPhaseShift00": gates.CPhaseShift00,
+    "CPhaseShift00.inv": gates.CPhaseShift00,
     "CPhaseShift01": gates.CPhaseShift01,
+    "CPhaseShift01.inv": gates.CPhaseShift01,
     "CPhaseShift10": gates.CPhaseShift10,
-    "RX": gates.Rx,
-    "RY": gates.Ry,
-    "RZ": gates.Rz,
-    "SWAP": gates.Swap,
-    "CSWAP": gates.CSwap,
+    "CPhaseShift10.inv": gates.CPhaseShift10,
     "ISWAP": gates.ISwap,
+    "ISWAP.inv": gates.PSwap,
     "PSWAP": gates.PSwap,
+    "PSWAP.inv": gates.PSwap,
     "XY": gates.XY,
+    "XY.inv": gates.XY,
     "XX": gates.XX,
+    "XX.inv": gates.XX,
     "YY": gates.YY,
+    "YY.inv": gates.YY,
     "ZZ": gates.ZZ,
-    "Toffoli": gates.CCNot,
-    "QubitUnitary": gates.Unitary,
+    "ZZ.inv": gates.ZZ,
```

Comment:
Using a giant dict seems a bit unwieldy, especially since there's a different entry for the inverses too. I'd suggest using [`singledispatch`](https://docs.python.org/3/library/functools.html#functools.singledispatch), the same way we do in the [default simulator](https://github.com/aws/amazon-braket-default-simulator-python/blob/main/src/braket/default_simulator/gate_operations.py#L68-L70). It might be more verbose, but this way, the inverses live with their base operations, and we can also avoid having specialized logic in the translator to deal with certain gates.

Reply:
agreed, I'll post an update using singledispatch